plugins/coverbrowser.koplugin/main.lua

```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
+                                UIManager:show( InfoMessage:new{ text = summary } )
+                            end)
+                        end
+                    })
+                end,
+            },
+            {
+                text = _("Delete cache database"),
+                keep_menu_open = true,
+                callback = function()
+                    local ConfirmBox = require("ui/widget/confirmbox")
+                    UIManager:close(self.file_dialog)
+                    UIManager:show(ConfirmBox:new{
+                        text = _("Are you sure that you want to delete cover and metadata cache?\n(This will also reset your library view settings.)"),
+                        ok_text = _("Purge"),
+                        ok_callback = function()
+                            BookInfoManager:deleteDb()
+                        end
+                    })
+                end,
+            },
+        },
+    })
+    -- the end
+    -------------------------------------------------------------
     menu_items.filemanager_display_mode = {
-        text = _("Display mode"),
+        text = _("Library view"),
         sub_item_table = sub_item_table,
     }
 
-    -- add Mosaic / Detailed list mode settings to File browser Settings submenu
-    -- next to Classic mode settings
-    if menu_items.filebrowser_settings == nil then return end
-    local fc = self.ui.file_chooser
-    table.insert (menu_items.filebrowser_settings.sub_item_table, 5, {
-        text = _("Mosaic and detailed list settings"),
+    table.insert(sub_item_table, {
+        text = _("Collections and history view"),
+        separator = true,
+        sub_item_table = {
+            {
+                text = _("Same as library view"),
+                checked_func = function()
+                    return BookInfoManager:getSetting("unified_display_mode")
+                end,
+                callback = function()
+                    if BookInfoManager:toggleSetting("unified_display_mode") then
+                        self:setupHistoryDisplayMode(filemanager_display_mode)
+                        self:setupCollectionDisplayMode(filemanager_display_mode)
+                    end
+                end,
+            },
+            {
+                text = _("History tab"),
+                enabled_func = function()
+                    return not BookInfoManager:getSetting("unified_display_mode")
+                end,
+                sub_item_table = history_sub_item_table,
+            },
+            {
+                text = _("Collections tab"),
+                enabled_func = function()
+                    return not BookInfoManager:getSetting("unified_display_mode")
+                end,
+                sub_item_table = collection_sub_item_table,
+                separator = true,
+            }
+        }
+    })
+
+    table.insert(sub_item_table, menu_items.filebrowser_settings.sub_item_table[5])
```

Comment:
A comment "Classic view properties" would be nice.

Reply:
will do